File: Data Structures & Algorithms/duplicate-integer/submission-0.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        set<int> a;
        for(int i =0;i<nums.size();i++){
            if(a.find(nums[i])!=a.end()){
                return true;
            }
            a.insert(nums[i]);
        }
        return false;
    }
};

File: Data Structures & Algorithms/non-cyclical-number/submission-0.py
class Solution:
    def isHappy(self, n: int) -> bool:
        m ={}
        while n != 1:
            s=0
            while n:
                s+=(n%10)**2
                n//=10
            if s in m:
                return False
            n=s
            m[n]=1
            
        return True